rllib/policy/dynamic_tf_policy_v2.py

```diff
@@ -710,9 +710,20 @@ def _initialize_loss_from_dummy_batch(
             self._input_dict[key] = get_placeholder(value=value, name=key)
             if key not in self.view_requirements:
                 logger.info("Adding extra-action-fetch `{}` to view-reqs.".format(key))
+
+                # Some models specify parameter space shape with
+                # TensorShape[Dimension(...)], which are not compatible with gym.
+                # Reformat the shape here into a sequence of ints.
+                shape = []
+                for dim in value.shape[1:]:
```

Comment:
Well, the code that I modified was already iterating only over `value.shape[1:]`, so I needed to keep it the same here to keep the same functionality. The real answer is that as far as I can tell the first dimension is always `None`, meaning that there is an unspecified number of elements along that dimension, whereas the other dimensions are restricted. I'm guessing this dimension is unspecified because it holds the number of samples in the dataset. But I think @gjoliver would be able to answer more completely here.